File: alyak-api-server/src/main/java/com/github/seungjae97/alyak/alyakapiserver/domain/notification/dto/request/UpsertDeviceTokenRequest.java
package com.github.seungjae97.alyak.alyakapiserver.domain.notification.dto.request;

import com.github.seungjae97.alyak.alyakapiserver.domain.notification.entity.DevicePlatform;
import lombok.Data;

@Data
public class UpsertDeviceTokenRequest {
    private String deviceId;
    private String fcmToken;
    private DevicePlatform platform;
    private String appVersion;
}

File: application_project/app/src/main/java/com/alyak/detector/feature/notification/data/model/UpsertDeviceTokenRequest.kt
package com.alyak.detector.feature.notification.data.model

import com.google.gson.annotations.SerializedName

data class UpsertDeviceTokenRequest(
    @SerializedName("deviceId")
    val deviceId: String,
    @SerializedName("fcmToken")
    val fcmToken: String,
    @SerializedName("platform")
    val platform: DevicePlatform,
    @SerializedName("appVersion")
    val appVersion: String? = null,
)
